lib/viewmodels/user_centauri_points_view_model.dart

```diff
@@ -0,0 +1,41 @@
+import "dart:async";
+
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+
+/// A view model that provides the number of centauri points of a user
+/// given its user id. Allow a null [arg] (user id) to return null, which
+/// is useful for loading states.
+class UserCentauriPointsViewModel
+    extends FamilyAsyncNotifier<int?, UserIdFirestore?> {
+  UserCentauriPointsViewModel();
+
+  @override
+  Future<int?> build(UserIdFirestore? arg) async {
+    final userDataBase = ref.watch(userRepositoryServiceProvider);
+
+    if (arg == null) {
+      return null;
+    }
+    final userID = arg;
+
+    final user = await userDataBase.getUser(userID);
+    final userCentauri = user.data.centauriPoints;
+    return userCentauri;
+  }
+
+  /// Refresh the number of centauri points of the user.
+  Future<void> refresh() async {
+    state = await AsyncValue.guard(() => build(arg));
+  }
+
+  /// Refresh the number of centauri points of the user.
+  void refreshWithCentauriPointsNumber(int centauriPoints) =>
+      state = AsyncValue.data(centauriPoints);
```

Comment:
After a discussion with @gruvw on a similar matter for my PR [#316](https://github.com/ProximaEPFL/proxima/issues/316), it appears that we may want to be careful with such small optimisations over Firestore's cache. It is probably better to be consistent everywhere, i.e. to trust the cache or to make optimisations everywhere.

Reply:
I did not know 👍🏼 

I decided to remove in [243304f](https://github.com/ProximaEPFL/proxima/pull/348/commits/243304f86b9f757c4460da135fb1ba639f882e5f) the `.refreshWithCentauriPointsNumber` as the` .refresh` have a more general usage.

Thank you 💯 

Tell me what you think 👍🏼

Reply:
This is perfect :)

(Don't worry, I did not know this morning either haha)